lib/matplotlib/_api/deprecation.py

```diff
@@ -390,12 +390,22 @@ def func(used_arg, other_arg, unused, more_args): ...
         is_varargs = kind is inspect.Parameter.VAR_POSITIONAL
         is_varkwargs = kind is inspect.Parameter.VAR_KEYWORD
         if not is_varargs and not is_varkwargs:
+            name_idx = (
+                # Deprecated parameter can't be passed positionally.
+                float("inf") if kind is inspect.Parameter.KEYWORD_ONLY
```

Comment:
Is str-to-float conversion slower, or faster than module access of `np.inf`?

Reply:
Nope: 

~~~
In [2]: %timeit np.inf
117 ns ± 0.63 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit float("inf")
294 ns ± 11.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
~~~

But that's really micro-optimization.

Reply:
we don't import np (or math) yet in that module, so I went for math.inf instead to stay in stdlib.